<a href="https://colab.research.google.com/github/yexf308/PracticalML/blob/main/Week_2_Case_Study_Video_Sale.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

We picked a dataset on video game sales and ratings from [Video Game Sales with Ratings](https://www.kaggle.com/datasets/rush4ratio/video-game-sales-with-ratings). Some of the games in this dataset lack ratings; so, let’s filter for only those examples that have all of their values present.

In [ ]:
!wget https://raw.githubusercontent.com/yexf308/PracticalML/main/data/video_games_sales.csv?raw=true -O video_games_sales.csv


--2023-05-17 17:44:48--  https://raw.githubusercontent.com/yexf308/PracticalML/main/data/video_games_sales.csv?raw=true
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1618040 (1.5M) [text/plain]
Saving to: ‘video_games_sales.csv’

video_games_sales.c 100%[===================>]   1.54M  --.-KB/s    in 0.04s   

2023-05-17 17:44:48 (37.6 MB/s) - ‘video_games_sales.csv’ saved [1618040/1618040]



In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

sns.set()

# Graphics in SVG format are more sharp and legible
%config InlineBackend.figure_format = 'svg'

# Increase the default plot size and set the color scheme
plt.rcParams["figure.figsize"] = (8, 5)
plt.rcParams["image.cmap"] = "viridis"

In [ ]:
# Load the data into a pandas dataframe and droped NA data entry
df = pd.read_csv("video_games_sales.csv").dropna()
print(df.shape)

(6825, 16)


In [ ]:
# Next, print the summary of the DataFrame to check data types and to verify everything is non-null.
print(df.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6825 entries, 0 to 16706
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             6825 non-null   object 
 1   Platform         6825 non-null   object 
 2   Year_of_Release  6825 non-null   float64
 3   Genre            6825 non-null   object 
 4   Publisher        6825 non-null   object 
 5   NA_Sales         6825 non-null   float64
 6   EU_Sales         6825 non-null   float64
 7   JP_Sales         6825 non-null   float64
 8   Other_Sales      6825 non-null   float64
 9   Global_Sales     6825 non-null   float64
 10  Critic_Score     6825 non-null   float64
 11  Critic_Count     6825 non-null   float64
 12  User_Score       6825 non-null   object 
 13  User_Count       6825 non-null   float64
 14  Developer        6825 non-null   object 
 15  Rating           6825 non-null   object 
dtypes: float64(9), object(7)
memory usage: 906.4+ KB
None


In [ ]:
df["Year_of_Release"]

0        2006.0
2        2008.0
3        2009.0
6        2006.0
7        2006.0
          ...  
16667    2001.0
16677    2002.0
16696    2014.0
16700    2011.0
16706    2011.0
Name: Year_of_Release, Length: 6825, dtype: float64

In [ ]:
df["User_Score"]

0          8
2        8.3
3          8
6        8.5
7        6.6
        ... 
16667    2.4
16677    8.8
16696    7.6
16700    5.8
16706    7.2
Name: User_Score, Length: 6825, dtype: object

## Task 1: 
We see that pandas has loaded some of the numerical features as object type. **We will explicitly convert those columns into float and int.**

In [ ]:
print(df.info())

The resulting DataFrame contains 6825 examples and 16 columns. Let’s look at the first few entries with the ```head()``` method to check that everything has been parsed correctly. 

In [ ]:

df.head()

Name Platform  Year_of_Release     Genre Publisher  \
0             Wii Sports      Wii             2006    Sports  Nintendo   
2         Mario Kart Wii      Wii             2008    Racing  Nintendo   
3      Wii Sports Resort      Wii             2009    Sports  Nintendo   
6  New Super Mario Bros.       DS             2006  Platform  Nintendo   
7               Wii Play      Wii             2006      Misc  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
6     11.28      9.14      6.50         2.88         29.80          89.0   
7     13.96      9.18      2.93         2.84         28.92          58.0   

   Critic_Count  User_Score  User_Count Developer Rating  
0            51         8.0         322  Nintendo      E  
2            73         8.3         709  Nintendo      E  
3            73         8.0         192  Nintendo      E  
6            65         8.5         431  Nintendo      E  
7            41         6.6         129  Nintendo      E

## Task 2:
**Create a plot of video game sales by country and year.**

In [ ]:
# or use matplotlib


## Task 3
**Create a pairwise relationships plot for Global_Sales, Critic_Score, Critic_Count, User_Score, User_Count.**


In [ ]:
%config InlineBackend.figure_format = 'png'


In [ ]:
%config InlineBackend.figure_format = 'svg'


## Task 4
**Plot a distribution of observations critics’ ratings**

## Task 5
To look more closely at the relationship between two numerical variables, **use jointplot to see how the Critic_Score and User_Score features are related.**



## Task 6
**Use Boxplot compare critics’ ratings for the top 5 biggest gaming platforms.**

## Task 7 
**Use heatmap to visualize the total sales of games by genre and gaming platform.**

